packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx

```diff
@@ -163,27 +166,28 @@ export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> =
       }
     case 'scatter':
       const isAreaChart = data.some((series: any) => series.fill === 'tonexty' || series.fill === 'tozeroy');
-      if (isXDate || isXNumber) {
+      const renderChart = (chartProps: any) => {
         if (isAreaChart) {
-          return (
-            <AreaChart
-              {...transformPlotlyJsonToScatterChartProps({ data, layout }, true, colorMap, isDarkTheme)}
-              legendProps={legendProps}
-              componentRef={chartRef}
-            />
-          );
+          return <AreaChart {...chartProps} />;
         }
-        return (
-          <LineChart
-            {...transformPlotlyJsonToScatterChartProps({ data, layout }, false, colorMap, isDarkTheme)}
-            legendProps={{
-              onChange: onActiveLegendsChange,
-              canSelectMultipleLegends: true,
-              selectedLegends: activeLegends,
-            }}
-            componentRef={chartRef}
-          />
-        );
+        return <LineChart {...chartProps} />;
+      };
+      if (isXDate || isXNumber) {
+        const chartProps = {
+          ...transformPlotlyJsonToScatterChartProps({ data, layout }, isAreaChart, colorMap, isDarkTheme),
+          legendProps,
+        };
+        return renderChart(chartProps);
+      } else if (isXMonth) {
+        const updatedData = data.map((dataPoint: any) => ({
+          ...dataPoint,
+          x: updateXValues(dataPoint.x),
+        }));
+        const chartProps = {
+          ...transformPlotlyJsonToScatterChartProps({ data: updatedData, layout }, isAreaChart, colorMap, isDarkTheme),
```

Comment:
>transformPlotlyJsonToScatterChartProps [](http://example.com/codeflow?start=13&length=38)

where are you handling different legendProps for line chart

Reply:
it can be handled in chartProps only, right? Also I missed componentRef={chartRef}, that also will add.

Reply:
add legend props to chartProps if it is line. This check will anyways go away in couple of days.